### PR TITLE
fix anonymous messages

### PR DESCRIPTION
**Pull request by nkeyes**

### CODE REVIEW

**Comment by Ferdy89** [comment]
@nkeyes I don't think I know enough about Philotic to properly understand/evaluate these changes, but code looks good and tests look thorough :+1:

File: lib/philotic/message.rb [hunk 1]
```diff
@@ -101,6 +104,10 @@ def metadata=(options)
 
     private
 
+     def _is_anonymous_message?
+       self.is_a?(Philotic::Message) && self.class.name.nil?
+     end
```
Comment (Ferdy89) [inline lib/philotic/message.rb]: Super small thing here: indentation looks a bit off

**Comment by nkeyes** [comment]
@Ferdy89:
```Ruby
Philotic::Message.publish(foo: 'FOO')
Philotic::Message.publish(bar: 'Bar')
```
The above was blowing up because the list of accessors is stored on the class level, Philotic::Message in this case.
When the second message was published, `Philotic::Message` still has `:foo` as an expected header.(from the first publish call)
The use on anonymous classes in the class-level method `Philotic::Message.publish` fixes this issue.

**Comment by kirillseva** [comment]
:+1:

File: spec/philotic/message_spec.rb [hunk 2]
```diff
@@ -96,23 +102,105 @@ class TestEvent < TestEventParent
...
+      end
+
+      describe 'in succession' do
+        it 'builds the proper routable accessors on succesive calls' do
+          allow(Philotic.connection).to receive(:publish)
+          Philotic::Message.publish message_1_headers, message_1_payload
+
+          expect(Philotic.connection).to receive(:publish) do |message|
+            expect(message.class.attr_routable_accessors).not_to include(*message_1_headers.keys)
+            expect(message.class.attr_routable_accessors).to include(*message_2_headers.keys)
```
Comment (Ferdy89) [inline spec/philotic/message_spec.rb]: :100: Gotcha!

**Pull request merged**